ops/docker/op-stack-go/Dockerfile

```diff
@@ -46,8 +46,15 @@ ARG TARGETARCH
 # Build the Go services, utilizing caches and share the many common packages.
 # The "id" defaults to the value of "target", the cache will thus be reused during this build.
 # "sharing" defaults to "shared", the cache will thus be available to other concurrent docker builds.
+
+# For now fetch the v1 cannon binary from the op-challenger image
+FROM --platform=$BUILDPLATFORM us-docker.pkg.dev/oplabs-tools-artifacts/images/op-challenger:v1.1.0 AS cannon-builder-0
+
 FROM --platform=$BUILDPLATFORM builder AS cannon-builder
-ARG CANNON_VERSION=v0.0.0
+# note: bump this CANNON_VERSION when the VM behavior changes
+ARG CANNON_VERSION=v1.0.0
+COPY --from=cannon-builder-0 /usr/local/bin/cannon ./cannon/multicannon/embeds/cannon-0
+COPY --from=cannon-builder-0 /usr/local/bin/cannon ./cannon/multicannon/embeds/cannon-1
```

Comment:
At this point there aren't any incompatible changes to cannon so shouldn't we be using the local build to get the current version?  When we actually make the incompatible change we should ensure we have backwards compatibility (ie we should do it as part of https://github.com/ethereum-optimism/optimism/pull/12050). Otherwise when we merge this we won't be using the current source code for any of the cannon implementations which would be very confusing.

Reply:
Actually I think `make cannon` just winds up overwriting the two versions we copied in.

Reply:
The `make cannon` command on L58 replaces the copied embeds with the current version. Which means we don't need to copy from `cannon-builder-0`. I did this mostly to illustrate how versioning should work here.

Tho since #12050 will replace this anyways, I'll just remove this rn.